src/proposal/mod.rs

```diff
@@ -0,0 +1,151 @@
+#[serde(crate = "near_sdk::serde")]
+#[borsh(crate = "near_sdk::borsh")]
+#[schemars(crate = "near_sdk::schemars")]
+pub struct Proposal {
+    pub id: ProposalId,
+    pub author_id: AccountId,
+    #[serde(
+        serialize_with = "u64_dec_format::serialize",
+        deserialize_with = "u64_dec_format::deserialize"
+    )]
+    pub social_db_post_block_height: BlockHeight,
+    pub snapshot: ProposalSnapshot,
+    // // Excludes the current snapshot itself.
+    pub snapshot_history: Vec<ProposalSnapshot>,
+}
+
+impl From<VersionedProposal> for Proposal {
+    fn from(vp: VersionedProposal) -> Self {
+        match vp {
+            VersionedProposal::V0(v0) => v0,
+        }
+    }
+}
+
+impl From<Proposal> for VersionedProposal {
+    fn from(p: Proposal) -> Self {
+        VersionedProposal::V0(p)
+    }
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(crate = "near_sdk::serde")]
+#[borsh(crate = "near_sdk::borsh")]
+#[schemars(crate = "near_sdk::schemars")]
+pub struct ProposalSnapshot {
+    pub editor_id: AccountId,
+    #[serde(
+        serialize_with = "u64_dec_format::serialize",
+        deserialize_with = "u64_dec_format::deserialize"
+    )]
+    pub timestamp: Timestamp,
+    pub labels: HashSet<PostTag>,
+    #[serde(flatten)]
+    pub body: VersionedProposalBody,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(crate = "near_sdk::serde")]
+#[borsh(crate = "near_sdk::borsh")]
+#[schemars(crate = "near_sdk::schemars")]
+pub struct ProposalBodyV0 {
+    pub name: String,
+    pub category: String,
+    pub summary: String,
+    pub description: String,
+    pub linked_proposals: Vec<ProposalId>,
+    #[serde(
+        serialize_with = "u32_dec_format::serialize",
+        deserialize_with = "u32_dec_format::deserialize"
+    )]
+    pub requested_sponsorship_amount: u32,
+    pub requested_sponsorship_token: SponsorshipToken,
+    pub receiver_account: AccountId,
+    pub requested_sponsor: AccountId,
+    pub supervisor: Option<AccountId>,
+    pub payouts: Vec<Base58CryptoHash>,
+    pub timeline: TimelineStatus,
+}
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(crate = "near_sdk::serde")]
+#[serde(tag = "proposal_body_version")]
+#[borsh(crate = "near_sdk::borsh")]
+#[schemars(crate = "near_sdk::schemars")]
+pub enum VersionedProposalBody {
+    V0(ProposalBodyV0),
+}
+
+impl From<VersionedProposalBody> for ProposalBodyV0 {
+    fn from(solution: VersionedProposalBody) -> Self {
+        match solution {
+            VersionedProposalBody::V0(v0) => v0,
+        }
+    }
+}
+
+impl VersionedProposalBody {
+    pub fn latest_version(self) -> ProposalBodyV0 {
+        self.into()
+    }
+}
+
+pub fn get_proposal_summary(proposal_body: VersionedProposalBody) -> String {
+    proposal_body.clone().latest_version().summary
+}
```

Comment:
It seems that this function is only used in a single place, so I would inline it there.